compiler/src/org.graalvm.compiler.loop/src/org/graalvm/compiler/loop/DerivedScaledInductionVariable.java

```diff
@@ -134,6 +134,34 @@ public void deleteUnusedNodes() {
         GraphUtil.tryKillUnused(scale);
     }
 
+    @Override
+    public boolean isConstantScale(InductionVariable ref) {
+        return super.isConstantScale(ref) || (scale.isConstant() && base.isConstantScale(ref));
```

Comment:
Sorry i was not very clear last time, i think we should do the same as in `DerivedOffsetInductionVariable` in this class: use the simple expressions rather than the "super" methods.

To avoid back-and-forth delays i'm anyway taking the PR in this state for further testing. In the meanwhile could you consider doing those extra changes?

Reply:
Sorry. I should have reviewed other similar uses. I just pushed an updated patch. Thanks for the careful review and for your help with this.